bin/identity.sh

```diff
@@ -0,0 +1,424 @@
+    }
+    if (isset($options['is_html_signature'])) {
+        validate_boolean($options['is_html_signature'], 'is signature HTML (H)'); 
+    }
+    if (isset($options['bcc_email'])) {
+        validate_email($options['bcc_email'], 'bcc email');
+    }
+    if (isset($options['reply_to_email'])) {
+        validate_email($options['reply_to_email'], 'reply-to email');
+    }
+    if (isset($options['is_default'])) {
+        validate_boolean($options['is_default'], 'is default identity (S)');
+        $setAsDefault = filter_var($options['is_default'], FILTER_VALIDATE_BOOLEAN);
+    } 
+
+    $new_identity['email'] = get_option_value($options, 'email', '', false, true, "Enter the email e.g. -e somemail@example.com");
+    $new_identity['name'] = get_option_value($options, 'name', '', false, true, "Enter the name of an identity e.g. -n 'John Smith'");
+    $new_identity['organization']  = get_option_value($options, 'organization', '', false, false);
+    $new_identity['signature'] = get_option_value($options, 'signature', '', false, false);
+    $new_identity['html_signature'] = get_option_value($options, 'is_html_signature', 0, true, false);
+    $new_identity['bcc'] = get_option_value($options, 'bcc_email', '', false, false);
+    $new_identity['reply-to'] = get_option_value($options, 'reply_to_email', '', false, false);
+
+    $host = rcube_utils::get_host($options);
+    $user = get_user($username, $host);
+
+    if (!$user) {
+        exit;
+    }
+
+    $id = $user->insert_identity($new_identity);
+
+    if ($setAsDefault) {
+        $user->set_default($id);
+    }
+
+    echo "Identity created successfully.\n";
+}
+
+function update_identity($options) {
+    if (count($options) === 1) {
+        echo "update\n";
+        echo "======\n\n";
+        echo "    Update an existing identity with provided values.\n\n\n";
+        echo "    Mandatory arguments:\n\n";
+        echo "    -u\n";
+        echo "        username - the identity holder e.g. -u mainmail@example.com\n\n";
+        echo "    -i\n";
+        echo "        identity id - the id of the identity being queried e.g. -i 70) use list sub-command to get identity id\n\n\n";
+        echo "    Optional arguments (at least one must be specified):\n\n";
+        echo "    -e\n";
+        echo "        email - the email of the identity e.g. identityemail@example.com\n\n";
+        echo "    -n\n";
+        echo "        the name of the identity - e.g. -n 'John Smith'\n\n";
+        echo "    -o\n";
+        echo "        organization - e.g. -o 'Your Organization Name'\n\n";
+        echo "    -r\n";
+        echo "        reply-to email - e.g. -r replytothisemail@example.com\n\n";
+        echo "    -b\n";
+        echo "        bcc email - e.g. -b bcc@example.com\n\n";
+        echo "    -s\n";
+        echo "        signature - e.g. -s 'Sincerely, John Smith'\n\n";
+        echo "    -H\n";
+        echo "        is signature HTML (empty value or 1 for yes, 0 for no) e.g. -H\n\n";
```

Comment:
I would approach it as setting HTML signature or plaintext signature, with a note that there can be only one at a time. Having a text with a separate flag seems too much implementation-flavored. Isn't it so that as an admin, I simply set a HTML signature, and not thinking of it as a text with the HTML flag.

Reply:
updated to use -h for html signature and -s for plain text, no need for flag any more